src/gdb/gdb.c

```diff
@@ -144,32 +145,74 @@ void gdb_handle_exception(void)
 void parse_request(void)
 {
 	unsigned char *request;
+	unsigned int i;
 	int addr;
-
-	while (1) {
-		request = get_packet();
-		/* Log any exception caused by debug exception */
-		gdb_debug_info(request);
-
-		/* Pick incoming request handler */
-		unsigned char command = *request++;
-
-		switch (command) {
-		/* Continue normal program execution and leave debug handler */
-		case 'c':
-			if (hex_to_int(&request, &addr))
-				sregs[DEBUG_PC] = addr;
-
-			/* return from exception */
-			return;
+	int length;
+
+while (1) {
+	request = get_packet();
+	/* Log any exception caused by debug exception */
+	gdb_debug_info(request);
+	/* Pick incoming request handler */
+	unsigned char command = *request++;
+
+	switch (command) {
+	/* Continue normal program execution and leave debug handler */
+	case 'c':
+		if (hex_to_int(&request, &addr))
+			sregs[DEBUG_PC] = addr;
+
+		/* return from exception */
+		return;
+	/* insert breakpoint */
+	case 'Z':
+		switch (*request++) {
+		/* HW breakpoint */
+		case '1':
+			if (*request++ == ',' && hex_to_int(&request, &addr) &&
+			    *request++ == ',' && hex_to_int(&request, &length)
+			    && *request == 0) {
+				for (i = 0; i < XCHAL_NUM_IBREAK; ++i) {
+					if (!(sregs[IBREAKENABLE] & (1 << i)) ||
+					sregs[IBREAKA + i] == addr) {
+						sregs[IBREAKA + i] = addr;
+						sregs[IBREAKENABLE] |= (1 << i);
+						write_sr(IBREAKA+i);
+						write_sr(IBREAKENABLE);
+						break;
+					}
+				}
+
+				if (i == XCHAL_NUM_IBREAK) {
+					strcpy((char *) remcom_out_buffer,
+					 "E02");
+				} else {
+					strcpy((char *)remcom_out_buffer, "OK");
+					sregs[INTENABLE]  &=
+					DISABLE_LOWER_INTERRUPTS_MASK;
+					write_sr(INTENABLE);
+				}
+			} else {
+				strcpy((char *)remcom_out_buffer, "E01");
+			}
```

Comment:
als please comment the above conditional blocks.

Reply:
E0X simply returns error code for GDB. This way GDB knows that this packet did not return valid information which can be used further.